k8-operator/controllers/infisicalsecret_helper.go

```diff
@@ -239,41 +291,65 @@ func (r *InfisicalSecretReconciler) ReconcileInfisicalSecret(ctx context.Context
 		secretVersionBasedOnETag = managedKubeSecret.Annotations[SECRET_VERSION_ANNOTATION]
 	}
 
+	if authStrategy == AuthStrategy.UNIVERSAL_MACHINE_IDENTITY && util.MachineIdentityTokenInstance == nil {
```

Comment:
nit: for util.MachineIdentityTokenInstance it might be better to place this global instance in this file since it isn't being used any where else